dashboard/app/controllers/api_controller.rb

```diff
@@ -336,8 +336,9 @@ def script_structure
   def user_progress
     if current_user
       script = Script.get_from_cache(params[:script])
-      user = params[:user_id] ? User.find(params[:user_id]) : current_user
-      render json: summarize_user_progress(script, user)
+      user = params[:user_id].present? ? User.find(params[:user_id]) : current_user
+      teacher_viewing_student = current_user.students.include?(user)
+      render json: summarize_user_progress(script, user).merge({teacherViewingStudent: teacher_viewing_student})
```

Comment:
I think it would be a bit more straightforward to include `teacherViewingStudent` in `def summarize_user_progress`, but it seems ok to punt on this if that's difficult for any reason.